ui/app/pages/settings/networks-tab/network-form/network-form.component.js

```diff
@@ -220,7 +222,11 @@ export default class NetworkForm extends PureComponent {
       const appendedRpc = `http://${url}`
       const validWhenAppended = validUrl.isWebUri(appendedRpc) && !url.match(/^https?:\/\/$/)
 
-      this.setErrorTo(stateKey, this.context.t(validWhenAppended ? 'uriErrorMsg' : invalidUrlErrorMsg))
+      this.setErrorTo(stateKey, this.context.t(validWhenAppended ? 'urlErrorMsg' : invalidUrlErrorMsg))
+    }
+
+    if (rpcUrls.includes(url)) {
```

Comment:
This validation function is used for both the RPC URL and the Block Explorer URL, but this particular error only applies to the former.

While it does seem doubtful anyone would set the block exporer URL to something that's in the RPC URL list, we should probably restrict this check to just the `rpcUrl` state key anyway.

Or we could split this validator into two, and put the common validation steps in a third function. Probably would be better if you wanted to spend the time doing that, as using this same validator for both seems error prone.